File: JunsBlog/Models/Comments/CommentSearchPagingOption.cs
using JunsBlog.Models.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace JunsBlog.Models.Comments
{
    public class CommentSearchPagingOption
    {
        public string SearchKey { get; set; }
        public SortOrderEnum SortOrder { get; set; }
        public SortByEnum SortBy { get; set; }
        public int CurrentPage { get; set; }
        public int PageSize { get; set; }
        public CommentSearchOnEnum SearchOn { get; set; }

        public CommentSearchPagingOption()
        {
            SearchKey = String.Empty;
            SortOrder = SortOrderEnum.Descending;
            SortBy = SortByEnum.UpdatedOn;
            SearchOn = CommentSearchOnEnum.CommentText;
            CurrentPage = 1;
            PageSize = 10;
        }
    }
}

File: JunsBlog/ClientApp/src/app/models/comment/commentSearchPagingOption.ts
import { SortByEnum } from "../enums/sortByEnum";
import { SortOrderEnum } from "../enums/sortOrderEnum";
import { commentSearchOnEnum } from "../enums/commentSearchOnEnum";

export class CommentSearchPagingOption {
    currentPage: number;
    pageSize: number;
    searchKey: string;
    sortBy: SortByEnum;
    sortOrder: SortOrderEnum;
    searchOn: commentSearchOnEnum;
    constructor(currentPage: number = 1, pageSize: number = 10, searchKey: string = null,   searchOn: commentSearchOnEnum = commentSearchOnEnum.CommentText,
         sortBy: SortByEnum = SortByEnum.CreatedOn, sortOrder: SortOrderEnum = SortOrderEnum.descending){
            this.currentPage = currentPage;
            this.pageSize = pageSize;
            this.searchKey = searchKey;
            this.sortBy = sortBy;
            this.sortOrder = sortOrder;
            this.searchOn = searchOn;
    }
}